### DISCUSSION TITLE
LinkNodes not working as expected for getSubstructMatches

### DISCUSSION THREAD

**Discussion by monktues** [rdkit/rdkit#6216] [Q&A] [answered]
Describe the bug: A clear and concise description of what the bug is.
- a substructure search using a CXSmarts with linker node of length one or two only finds nodes with one atom

To Reproduce Steps to reproduce the behavior, feel free to paste a bit of Python in here.
=====FIRST CODE FROM JUPYTER NOTEBOOK CELL THAT WORKS AS EXPECTED=====
```
from rdkit import Chem
# phenyl and pyridyl connected by oxygen
m = Chem.MolFromSmiles('O(C1=CC=CC=C1)C1=CC=NC=C1')

# query with linknode specifying that the node has 1 or 2 repetitions
qsmarts = Chem.MolFromSmarts('*(-c1ccccc1)-c1ccncc1 |LN:0:1.2|')

# find substructure in molecule
matches = m.GetSubstructMatches(qsmarts)
# show atom indices matching substructure
print(matches) # prints: ((0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12),)
# show molecule with highlighted match found
m
```
=====SECOND JUPYTER NOTEBOOK CELL CODE THAT DOES NOT WORK AS EXPECTED=====
```
# phenyl and pyridyl connected by ethylene spacer (2 atoms between aryl rings)
m = Chem.MolFromSmiles('C(CC1=CC=NC=C1)C1=CC=CC=C1')

# find substructure in molecule
matches = m.GetSubstructMatches(qsmarts)
# show atom indices matching substructure
print(matches)
# this prints "()", i.e. no matches
```
Expected behavior A clear and concise description of what you expected to happen.
- I expected the CXSmarts code "LN:0:1.2" to match either one or two atoms between the aromatic rings
- If I change the code to "LN:0:2.2" it still matches the single oxygen atom but not the two carbon atoms; it does not seem to be reading the minimum and maximum repetitions

Screenshots If applicable, add screenshots to help explain your problem.
<img width="708" alt="image" src="https://user-images.githubusercontent.com/67247390/226464086-a2e8bf8b-c4b6-4528-b740-e2c8dce9aa38.png">
<img width="710" alt="image" src="https://user-images.githubusercontent.com/67247390/226464219-19b2d28b-ef69-451b-a43c-fb045e6161c3.png">
<img width="775" alt="image" src="https://user-images.githubusercontent.com/67247390/226464391-707d130a-e522-45bd-b8ac-5328e78a78b5.png">

Configuration (please complete the following information):

RDKit version:  2022.09.4
OS: Mac OS Monterey 12.1
Python version (if relevant):  3.9.15
Are you using conda? yes
If you are using conda, which channel did you install the rdkit from? conda-forge
If you are not using conda: how did you install the RDKit?

Additional context Add any other context about the problem here.
- In general, I'm trying to use rdkit to do R-group deconvolution with both R-groups and LinkNodes
- Thanks in advance for any help you can provide

**Comment by greglandrum** [comment, marked as answer]
Hi @monktues. There's another step you have to take in order to have linknodes used in substructure searches. It's explained in this blog post:
https://greglandrum.github.io/rdkit-blog/posts/2021-08-03-generalized-substructure-search.html

**Reply by monktues** [reply]
Thanks, Greg!

For others, I only needed to add five new lines to get it to work!  See below for the new lines and my understanding.

```
from rdkit import Chem

# phenyl and pyridyl connected by ethylene spacer (2 atoms between aryl rings)
m = Chem.MolFromSmiles('C(CC1=CC=NC=C1)C1=CC=CC=C1')

# query with linknode specifying that the node has 1 or 2 repetitions
qsmarts = Chem.MolFromSmarts('*(-c1ccccc1)-c1ccncc1 |LN:0:1.2|')

# --------------NEW LINES--------------
# enumerate the different linker lengths
from rdkit.Chem import rdMolEnumerator
qry_bundle = rdMolEnumerator.Enumerate(qsmarts)

# apply AdjustQueryProperties because I'm using a wildcard dummy atom *
newq = Chem.MolBundle()
for q in qry_bundle:
    newq.AddMol(Chem.AdjustQueryProperties(q))
# ----------END NEW LINES--------------

# find substructure in molecule
#matches = m.GetSubstructMatches(qsmarts)
matches = m.GetSubstructMatches(newq) # changed argument to new MolBundle newq

# show atom indices matching substructure
print(matches) # prints a non-empty result which means it matched the substructure
```
